single_stage_detector/ssd/mlcube.py

```diff
@@ -0,0 +1,99 @@
+import os
+import yaml
+import typer
+import shutil
+import subprocess
+from pathlib import Path
+
+
+app = typer.Typer()
+
+
+class DownloadDataTask(object):
+
+    urls = {
+        "train2017": "http://images.cocodataset.org/zips/train2017.zip",
+        "val2017": "http://images.cocodataset.org/zips/val2017.zip",
+        "annotations": "http://images.cocodataset.org/annotations/annotations_trainval2017.zip"
+    }
+
+    @staticmethod
+    def download(cache_dir: Path, data_dir: Path, name: str) -> None:
+        os.makedirs(cache_dir, exist_ok=True)
+        os.makedirs(data_dir, exist_ok=True)
+
+        dest_dir = data_dir / name
+        if dest_dir.exists():
+            print(f"Data ({name}) has already been downloaded: {dest_dir}")
+            return
+
+        archive_name = Path(DownloadDataTask.urls[name]).name
+        cached_archive = cache_dir / archive_name
+        if not cached_archive.exists():
+            print(f"Data ({name}) is not in cache ({cached_archive}), downloading ...")
+            os.system(f"cd {cache_dir}; curl -O {DownloadDataTask.urls[name]};")
```

Comment:
this should be filed by an md5sum check (and we should record the md5sum for each downloaded file in addition to just the url)